### PR TITLE
Mastery and item parsing

### PR DESCRIPTION
**Pull request by sathorn6**

### CODE REVIEW

File: src/libOL/ItemEntry.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include <cassert>
+#include <cstring>
+
+namespace libol {
+    ItemEntry ItemEntry::decode(std::ifstream& ifs) {
+        ItemEntry entry;
+
+        ifs.read(reinterpret_cast<char *>(&entry.itemId), sizeof(entry.itemId));
+        assert(ifs.get() == 0x00);
+        assert(ifs.get() == 0x00);
```
Comment (lukegb) [inline src/libOL/ItemEntry.cpp]: Did you mean to stick this in twice?

File: src/libOL_test/main.cpp [hunk 2]
```diff
@@ -21,13 +21,33 @@ int test_keyframe(std::vector<std::string> arguments)
     std::ifstream ifs(arguments.at(0), std::ios::binary);
     if (!ifs) {
         std::cerr << "Failed to open " << arguments.at(0) << ": " << strerror(errno) << std::endl;
-	return 2;
+        return 2;
```
Comment (lukegb) [inline src/libOL_test/main.cpp:24]: Oops, nice catch :+1:

File: src/libOL/ItemEntry.cpp (hunk 1 above)
Comment (toulouse) [inline src/libOL/ItemEntry.cpp]: Probably, since it's checking for two 0x00 bytes, no?

Comment (lukegb) [inline src/libOL/ItemEntry.cpp]: Herpderp.

This is probably not a good way of checking, though. assert statements are not guaranteed to be executed - if NDEBUG is defined, assert statements will be no-opped.

Comment (toulouse) [inline src/libOL/ItemEntry.cpp]: Agreed. I'd probably just read  them even if asserts were off, then assert the saved value.

Comment (sathorn6) [inline src/libOL/ItemEntry.cpp]: But i think these checks are part of the parser and not just a debug feature, right? Then they should probably not be done with assert in the first place.

Comment (toulouse) [inline src/libOL/ItemEntry.cpp]: That's what I was saying -- something like this:

```
uint8_t null0 = ifs.get();
assert(null0 == 0x00);
uint8_t null1 = ifs.get();
assert(null1 == 0x00);
```

Comment (lukegb) [inline src/libOL/ItemEntry.cpp]: (I think) what @satan6 was pointing out is that maybe they shouldn't be done with `assert` at all. I'm not sure I agree... I think that while they could be done with assert, the code shouldn't _depend_ on `NDEBUG` not being defined. The _check_ can depend on it, but not the functioning of the code.

Comment (sathorn6) [inline src/libOL/ItemEntry.cpp]: Yeah i meant that the parser is supposed to always tell you if something is not right. So the comparison itself should also not depend on some debug flag. But i'm not sure about that, so i just fixed it like suggested for now.

**Pull request merged**